### PR TITLE
Add support for shortcut files

### PR DESCRIPTION
**Pull request by gave92**
This PR adds support for shortcut files (".lnk", ".url")
Related issue: #177

**What works**
- Listing shortcuts with thumbnail
- Opening file shortcuts
- Navigating to destination when opening a shortcut to a folder (also open in new tab/window)
- Deleting shortcuts (but can't send to recycle bin)
- Copy/Move shortcut files
- Renaming shortcuts
- Dropping a file inside a folder shortcut

**What doesn't work**
- Drag shortcuts --> *I have no clue how to make this work*

**What's missing**
- ~Properties window could allow to edit shortcuts~
- ~Creating new shortcuts~ (from context menu)
- ~Share a .url file as link~
- ~A decent icon on lnk/url files~ --> *thanks @XPoppyX!*

*Edit*

@duke7553 We are now using 3 separate methods for supporting all file operations:
- StorageFile API
- UWP ***FromApp methods
- Win32 functions in the fulltrust process

This leads to a bit of complication, also the `ItemViewModel` starts to look a bit messy/hackish. What do you think?
Btw have you already made some progress on #632?

### CODE REVIEW

**Comment by Jaiganeshkumaran** [comment]
@gave92 You can't share a shortcut file even in Explorer but for .URL files we could share the link of the shortcut.

**Comment by Jaiganeshkumaran** [comment]
Is is possible to create new shortcut files with this PR?

**Comment by gave92** [comment]
@Jaiganeshkumaran I have yet to look into creating new shortcuts, currently you can only view and use existing ones. Ideally you should be able to create new shortcuts and edit existing ones in the Property window.

**Comment by Jaiganeshkumaran** [comment]
@gave92 Don't share anything for LNK files since they mostly contain a local path and sharing it doesn't make sense unless the path is accessible from other PCs.

**Comment by gave92** [comment]
@Jaiganeshkumaran now you can create shortcuts to files and folders from right-click context menu :)

**Comment by Jaiganeshkumaran** [comment]
Can you create a shortcut from a selected folder? Please add an option to create shortcut from selected folder.

**Comment by gave92** [comment]
@Jaiganeshkumaran yes you can already do that

**Comment by gave92** [comment]
Would you consider something like this for the Property windows?

image

**Comment by duke7553** [comment]
@gave92 Sorry I couldn't get to this one earlier. 

To answer your questions:
- We would not want to use a TabView for Static navigation. A Pivot-like control is probably more suited for this scenario.
- Our goal is to switch over to exclusively use the *FromApp methods for File operations, which I haven't got around to yet.

Thank you for your work on this.

**Comment by yaichenbaum** [comment]
> Would you consider something like this for the Property windows?
> 
> 

As @duke7553 mentioned, we would probably want to use something like a horizontal navigation menu here.

**Comment by gave92** [comment]
@yair100 So something like [Pivot](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/pivot) or [horizontal NavigationView](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/navigationview) correct?

**Comment by XPoppyX** [comment]
@gave92 what about this icon?

Anotación 2020-07-26 122526

**Comment by yair100** [comment]
> @yair100 So something like [Pivot](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/pivot) or [horizontal NavigationView](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/navigationview) correct?

If I recall correctly, it's recommended to use the NavigationView over the Pivot control.

**Comment by XPoppyX** [comment]
steam game shortcuts are shown with this icon, instead of the game icon
Anotación 2020-07-26 125627

**Comment by gave92** [comment]
@XPoppyX I can confirm that at the moment ".url" files won't display the proper icon

**Comment by XPoppyX** [comment]
I have tested the new icon but with reduced size:

**GridView:** 28 => 18

**TilesView:** 22 => 15

Anotación 2020-07-26 130725

I think the current size is somewhat big

**Comment by XPoppyX** [comment]
@gave92 Check your repository, I added a PR

**Comment by gave92** [comment]
@XPoppyX thanks a lot!

**Comment by XPoppyX** [comment]
what about md5hash?
Anotación 2020-07-26 144251

**Comment by gave92** [comment]
I take it doesn't work? Only on shortcuts or everywhere?

**Comment by XPoppyX** [comment]
only on shortcuts

**Comment by gave92** [comment]
@XPoppyX should be fixed, note that it will be the hash of the shortcut `target` and not of the shortcut itself

**Comment by XPoppyX** [comment]
icon proposal
Anotación 2020-07-26 171619

**Comment by XPoppyX** [comment]
- in "Shortcut" tab, for exe files it should display as "Application" type instead of just "file"
- in "General" tab, the type does not display correctly in my language:
Anotación 2020-07-26 185500
_this also happens in tiles view, see:_ #843
- Is it okay to display the type as "Ink File" instead of "Shortcut (.Ink)"?

**Comment by Jaiganeshkumaran** [comment]
> * Is it okay to display the type as "Ink File" instead of "Shortcut (.Ink)"?

No, it's not. Not everyone knows what the INK extension means.

**Comment by gave92** [comment]
@XPoppyX most of your notes should now be fixed:
- icon proposal --> done!
- in "Shortcut" tab, for exe files it should display as "Application" type instead of just "file" --> done!
- in "General" tab, the type does not display correctly in my language: --> done (needs resource translation/only for shortcuts)
- Is it okay to display the type as "Ink File" instead of "Shortcut (.Ink)"? --> done (needs resource translation)

**Comment by XPoppyX** [comment]
nice

File: Files/Views/Pages/Properties.xaml.cs [hunk 1]
```diff
@@ -78,11 +51,7 @@ private void Properties_Loaded(object sender, RoutedEventArgs e)
                 TitleBar.ButtonInactiveBackgroundColor = Colors.Transparent;
                 AppSettings.UpdateThemeElements.Execute(null);
             }
-
-            if (BaseProperties != null)
-            {
-                BaseProperties.GetSpecialProperties();
-            }
+            AppSettings.ThemeModeChanged += AppSettings_ThemeModeChanged;
```
Comment (tsvietOK) [inline Files/Views/Pages/Properties.xaml.cs:54]: To fix black window controls (minimize, maximize, close) move `AppSettings.ThemeModeChanged += AppSettings_ThemeModeChanged;` before if in this method.

**Comment by tsvietOK** [comment]
Can you increase Properties window height from 475 to 550?

**Review by tsvietOK — CHANGES_REQUESTED** [review]
Disable MD5 has calculation for link item, as it is hash of the real file, not the link file.

**Comment by tsvietOK** [comment]
@gave92 Why did you removed "Created" and "Accessed" fields for link files?

**Comment by gave92** [comment]
@tsvietOK both were of the link target and not of the link itself (same as the md5 hash)

**Comment by tsvietOK** [comment]
Is it possible to to don't display link file extension independently from "Show file extension" setting?

File: Files/Interacts/Interaction.cs [hunk 2]
```diff
@@ -600,13 +633,38 @@ private async void Manager_DataRequested(DataTransferManager sender, DataRequest
...
+                {
+                    var value = new ValueSet
+                    {
+                        { "Arguments", "FileOperation" },
+                        { "fileop", "CreateLink" },
+                        { "targetpath", selectedItem.ItemPath },
+                        { "arguments", "" },
+                        { "workingdir", "" },
+                        { "runasadmin", false },
+                        { "filepath", Path.Combine(CurrentInstance.FilesystemViewModel.WorkingDirectory, selectedItem.ItemName + " - shortcut" + ".lnk") }
```
Comment (tsvietOK) [inline Files/Interacts/Interaction.cs:661]: Add ability to translate " - shortcut"

**Review by tsvietOK — CHANGES_REQUESTED** [review]

**Review by tsvietOK — DISMISSED** [review]

**Review by yair100 — CHANGES_REQUESTED** [review]

**Review by yair100 — APPROVED** [review]

**Pull request merged**